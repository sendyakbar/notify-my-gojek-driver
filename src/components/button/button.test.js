import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Button from './button.jsx'

describe('<Button> component', () => {
    const wrapper = shallow(<Button />)
    test('renders correctly', () => {
        expect(wrapper).toHaveLength(1)
    })
    test('it has a div', () => {
        const divWrapper = wrapper.find('div')
        expect(divWrapper).toHaveLength(1)
    })
    test('it has a button', () => {
        const buttonWrapper = wrapper.find('button')
        expect(buttonWrapper).toHaveLength(1)
    })
    test('renders correctly and matched snapshot', () => {
        const tree = renderer.create(<Button />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})