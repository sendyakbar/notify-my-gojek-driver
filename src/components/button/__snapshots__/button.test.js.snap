// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button> component renders correctly and matched snapshot 1`] = `
<div>
  <button
    className="button"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": undefined,
      }
    }
  />
</div>
`;
